dbt/adapters/spark/session.py

```diff
@@ -24,9 +24,10 @@ class Cursor:
     https://github.com/mkleehammer/pyodbc/wiki/Cursor
     """
 
-    def __init__(self) -> None:
+    def __init__(self, *, server_side_parameters: Optional[Dict[str, Any]] = None) -> None:
```

Comment:
I'm unfamiliar with this mechanic. What does the `*,` do in the signature? Is it similar to unpacking with `my_arg, *_`? Do we need it?

Reply:
Yes, it is similar to the unpacking you mentioned. I guess that in a signature this mechanic is called "packing" as it is the opposite of unpacking. (I could not quickly find a PEP about this.)

What it does: it packs all positional argument. In this case, all positional arguments are packed in nothing. If you add a parameter after the `*` you pack all positional arguments into that parameter (as a tuple):

```
def sum(*numbers):
    total = 0
    for number in number:
        total += umber
    return total
    
sum(1, 2, 3)
sum(5, 7, 9, 11)
```

The trick here is that the `*` without an argument forces all arguments after it to become key-word arguments. I like to use that to improve readability:

```
Connection(foo)
```

vs

```
Connection(server_side_parameters=foo)
```

The later is more readable.

`foo` is of course a badly chosen variable name. But, I expect the first positional argument of a `Connection` to be a `connection_string`, like `conn = pyodbc.connect(connection_str, autocommit=True)`, or something other than `server_side_paramters` which is an (optional) additional parameter.